client/client_test.go

```diff
@@ -43,16 +43,42 @@ func TestAddLeaf(t *testing.T) {
 		t.Errorf("Failed to create log: %v", err)
 	}
 
-	client := New(logID, env.ClientConn, testonly.Hasher)
-	client.MaxTries = 1
+	cli := trillian.NewTrillianLogClient(env.ClientConn)
+	for _, test := range []struct {
+		desc    string
+		client  trillian.TrillianLogClient
+		wantErr bool
+	}{
+		{
+			desc:   "success 1",
+			client: &MockLogClient{c: cli},
+		},
+		{
+			desc:   "success 2",
+			client: &MockLogClient{c: cli},
+		},
+		{
+			desc:    "invalid inclusion proof",
+			client:  &MockLogClient{c: cli, mGetInclusionProof: true},
+			wantErr: true,
+		},
+		{
+			desc:    "invalid consistency proof",
+			client:  &MockLogClient{c: cli, mGetConsistencyProof: true},
+			wantErr: true,
+		},
+	} {
+		client := New(logID, test.client, testonly.Hasher)
+		client.MaxTries = 1
 
-	if err, want := client.AddLeaf(ctx, []byte("foo")), codes.DeadlineExceeded; grpc.Code(err) != want {
-		t.Errorf("AddLeaf(): %v, want, %v", err, want)
-	}
-	env.Sequencer.OperationLoop() // Sequence the new node.
-	glog.Infof("try AddLeaf again")
-	if err := client.AddLeaf(ctx, []byte("foo")); err != nil {
-		t.Errorf("Failed to add Leaf: %v", err)
+		if err, want := client.AddLeaf(ctx, []byte(test.desc)), codes.DeadlineExceeded; grpc.Code(err) != want {
+			t.Errorf("AddLeaf(%v): %v, want, %v", test.desc, err, want)
```

Comment:
Add a `continue` after t.Errorf? It doesn't like the current test could go on after this.